app/controllers/spree/affirm_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Spree
+  class AffirmController < Spree::StoreController
+    protect_from_forgery except: [:confirm]
+
+    def confirm
+      order = current_order || raise(ActiveRecord::RecordNotFound)
```

Comment:
Can we do something else other than just redirecting the user to a 404 not found page if this happens?